lib/core/src/Cardano/Wallet.hs

```diff
@@ -765,3 +768,25 @@ newWalletLayer tracer bp db nw tl = do
                         return $ checkPassphrase pwd hpwd
                     return xprv
         action xprv
+
+
+{---------------------------------------------------------------------------
+                                 Helpers
+  Defined outside 'WalletLayer' since they don't benefit from being "layered".
+---------------------------------------------------------------------------}
+
+-- | Regularly poll the wallet to monitor it's syncing progress. Block until the
+-- wallet reaches 100%.
+waitForWalletSync
+    :: WalletLayer s t
+    -> Trace IO Text
+    -> WalletId
+    -> IO ()
+waitForWalletSync walletLayer tracer wid = do
+    (_, meta) <- unsafeRunExceptT $ readWallet walletLayer wid
```

Comment:
Why is this used in its `unsafe` form ? If it's now in the `core` package, we do want to careful with how this is used...

Reply:
👍 I made it return `ExceptT` now.